src/CoreMatchers/Descriptions/SelfDescribingValueEnumerable.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace JPopadak.CoreMatchers.Descriptions
+{
+    public class SelfDescribingValueEnumerable<T> : IEnumerable<IDescribable>
+    {
+        private readonly List<SelfDescribingValue<T>> values;
+
+        public SelfDescribingValueEnumerable(IEnumerable<T> values)
+        {
+            List<SelfDescribingValue<T>> describingValues = new List<SelfDescribingValue<T>>();
+            foreach (T value in values)
+            {
+                describingValues.Add(new SelfDescribingValue<T>(value));
+            }
+            this.values = describingValues;
+        }
+
+        public IEnumerator<IDescribable> GetEnumerator()
+        {
+            return values.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
```

Comment:
This is called implementing an interface explicitly. It is required by some interfaces.
http://stackoverflow.com/a/143423/6079304